iceoryx_utils/include/iceoryx_utils/internal/cxx/forward_list.inl

```diff
@@ -0,0 +1,645 @@
+}
+
+template <typename T, uint64_t Capacity>
+typename forward_list<T, Capacity>::iterator forward_list<T, Capacity>::insert_after(const_iterator citer,
+                                                                                     T&& data) noexcept
+{
+    return emplace_after(citer, std::forward<T>(data));
+}
+
+
+template <typename T, uint64_t Capacity>
+void forward_list<T, Capacity>::clear() noexcept
+{
+    for (auto iter = before_begin(); cbegin() != cend();)
+    {
+        erase_after(iter);
+        iter = before_begin();
+    }
+}
+
+
+/*************************/
+// iterator
+
+
+template <typename T, uint64_t Capacity>
+template <bool is_const_iter>
+forward_list<T, Capacity>::iterator_base<is_const_iter>::iterator_base(parentListPointer parent, size_type idx) noexcept
+    : m_list(parent)
+    , m_iterListNodeIdx(idx)
+{
+}
+
+template <typename T, uint64_t Capacity>
+template <bool is_const_iter>
+forward_list<T, Capacity>::iterator_base<is_const_iter>::iterator_base(const iterator_base<false>& iter)
+    : m_list(iter.m_list)
+    , m_iterListNodeIdx(iter.m_iterListNodeIdx)
+{
+}
+
+template <typename T, uint64_t Capacity>
+template <bool is_const_iter>
+auto forward_list<T, Capacity>::iterator_base<is_const_iter>::operator++() noexcept ->
+    typename forward_list<T, Capacity>::template iterator_base<is_const_iter>&
+{
+    if (!m_list->handleInvalidIterator(*this))
+    {
+        m_iterListNodeIdx = m_list->getNextIdx(m_iterListNodeIdx);
+    }
+    return *this;
+}
+
+
+template <typename T, uint64_t Capacity>
+template <bool is_const_iter>
+template <bool is_const_iter_other>
+bool forward_list<T, Capacity>::iterator_base<is_const_iter>::operator==(
+    const forward_list<T, Capacity>::iterator_base<is_const_iter_other>& rhs) const noexcept
+{
+    if (m_list->invalidIterOrDifferentLists(rhs) || m_list->handleInvalidIterator(*this))
+    {
+        return false;
+    }
+    // index comparison
+    return (m_iterListNodeIdx == rhs.m_iterListNodeIdx);
+}
+
+template <typename T, uint64_t Capacity>
+template <bool is_const_iter>
+template <bool is_const_iter_other>
+bool forward_list<T, Capacity>::iterator_base<is_const_iter>::operator!=(
+    const forward_list<T, Capacity>::iterator_base<is_const_iter_other>& rhs) const noexcept
+{
+    return !operator==(rhs);
+}
+
+
+template <typename T, uint64_t Capacity>
+template <bool is_const_iter>
+auto forward_list<T, Capacity>::iterator_base<is_const_iter>::operator*() const noexcept ->
+    typename forward_list<T, Capacity>::template iterator_base<is_const_iter>::reference
+{
+    return *operator->();
+}
+
+
+template <typename T, uint64_t Capacity>
+template <bool is_const_iter>
+auto forward_list<T, Capacity>::iterator_base<is_const_iter>::operator-> () const noexcept ->
+    typename forward_list<T, Capacity>::template iterator_base<is_const_iter>::pointer
+{
+    return m_list->getDataPtrFromIdx(m_iterListNodeIdx);
+}
+
+// iterator
+/*************************/
```

Comment:
It's just nitpicking that these separator lines are somehow weird. But i know that's for better readibility. Maybe just add some spacing and then just a comment like you did without these separator lines is sufficient here. But as  @elBoberido  would say: I have no strong feelings about this. ;-)